TikTokApi/tiktok.py

```diff
@@ -491,6 +499,53 @@ def byHashtag(self, hashtag, count=30, language='en', proxy=None, region='US') -
 
         return response[:count]
 
+    def getHashtagPager(self, hashtag, page_size=30, max_pages=0, proxy=None, language='en', region='US'):
+        id = self.getHashtagObject(hashtag)['challengeInfo']['challenge']['id']
+
+        before = 0
+        while True:
+            resp = self.hashtagPage(
+                id, page_size=page_size, before=before, proxy=proxy, language=language, region=region
+            )
+
+            has_more = resp['body']['hasMore']
+            try:
+                page = resp['body']['itemListData']
+            except KeyError:
+                # No mo results
+                page = []
+                has_more = False
+
+            before = resp['body']['maxCursor']
+
+            yield page
+            max_pages -= 1
+
+            if max_pages == 0:
+                has_more = False
```

Comment:
This is pretty clever, if I do say so myself.  If `max_pages` is set to a positive number, it will stop paging when it reaches exactly zero. but if it's set to zero or negative, it'll always go down and never exactly equal zero, so it'll just page until tiktok says no more :)